### ISSUE TITLE
Get all edges on a given node

### ISSUE THREAD

**Issue by leek2** [pydot/pydot#260]
I'm trying to walk a dot graph I read in, and I can't find any way to ask for a list of all edges (or all in edges, or out edges) on a given node.  If I know there is an edge from A->B, I can: graph.get_edge("A", "B").  But there doesn't seem to be anyway to ask for "All out edges on A."  I tried a variety of things like this, but just got an empty list: graph.get_edge("A")

Right now I've just made my own data structures to work around this, but it seems like an odd oversight to me.  Maybe I'm missing something?

**Comment by felix-hilden** [comment]
I'm not sure about the official approach of Pydot to this, but these sorts of things were my reason to using NetworkX, which seems to be highly compatible with Pydot. So in essence: analysis with NetworkX, dotfile and Graphviz operations with Pydot. In my humble opinion it's fine for Pydot to remain "just" an interface to Graphviz.

**Comment by sahansk2** [comment]
In the Python interpreter, running `help()` on a `Dot` object shows that there is indeed a function to get all edges: `get_edges()`. Might this be what you want, @leek2 ?

**Comment by leek2** [comment]
sahansk2, thank you for the suggestion, but no.  That gets all edges on the graph.  I want to be able to ask for just the edges on a given node.  Of course I can extract that from all the edges, but it would be nice if pydot had a more straight-forward way to do it.

**Comment by sahansk2** [comment]
Oh, sorry for misreading your question. Then, I haven't seen anything in pydot to directly support this functionality, which is a real shame.

**Comment by atharvakadlag** [comment]
If there is no prewritten function for it, you could do it manually
You could do that by checking get_edge with all the nodes via get_nodes
wouldnt be that difficult to write
ping me if you need any help writing it

**Comment by lkk7** [comment]
I don't think this will be done in the nearest future (pydot is just a graphviz wrapper).
